### PR TITLE
Fix timer trigger regression

### PR DESCRIPTION
**Pull request by manasdk**
* Trigger in timer trigger is a dict so access properties as a dict.
* Update tests to reflect the internals

### CODE REVIEW

**Comment by Kami** [comment]
This will also go into v0.13.1, right?

**Comment by manasdk** [comment]
> This will also go into v0.13.1, right?
yes

File: st2reactor/st2reactor/timer/base.py [hunk 1]
```diff
@@ -132,8 +134,9 @@ def _emit_trigger_instance(self, trigger):
             'executed_at': str(utc_now),
             'schedule': trigger['parameters'].get('time')
         }
+
         trace_context = TraceContext(trace_tag='%s-%s' % (self._get_trigger_type_name(trigger),
-                                                          trigger.name))
+                                                          trigger.get('name', uuid.uuid4().hex)))
```
Comment (Kami) [inline st2reactor/st2reactor/timer/base.py]: Is it possible to have a trigger without name at this point?

We randomly generate one if one is not provided so it seems like this shouldn't be possible.

**Comment by Kami** [comment]
LGTM

File: st2reactor/st2reactor/timer/base.py (hunk 1 above)
Comment (manasdk) [inline st2reactor/st2reactor/timer/base.py]: No. not possible. 

This trigger thing is such a disaster that I would hate for it to blow up for dumb things.

**Pull request merged**